src/administrative-sdk/category/category-controller.js

```diff
@@ -0,0 +1,117 @@
+import Category from './category';
+
+/**
+ * Controller class for the Category model.
+ * @private
+ */
+export default class CategoryController {
+  /**
+   * @param {Connection} connection - Object to use for making a connection to the REST API and Websocket server.
+   */
+  constructor(connection) {
+    /**
+     * Object to use for making a connection to the REST API and Websocket server.
+     * @type {Connection}
+     */
+    this._connection = connection;
+  }
+
+  /**
+   * Create a category.
+   *
+   * @param {Category} category - Object to create.
+   * @returns {Promise.<Category>} Promise containing the newly created Category.
+   * @throws {Promise.<Error>} category parameter of type "Category" is required.
+   * @throws {Promise.<Error>} If the server returned an error.
+   */
+  createCategory(category) {
+    if (!(category instanceof Category)) {
+      return Promise.reject(new Error('category parameter of type "Category" is required'));
+    }
+
+    const url = this._connection._settings.apiUrl + '/categories';
+    const fd = JSON.stringify(category);
+    return this._connection._secureAjaxPost(url, fd)
+      .then(data => {
+        const result = new Category(data.id, data.parent, data.name, data.description, data.color,
```

Comment:
Consider this:

```js
// category-controller.js, creating new Category object
someAjaxPost(url, data).then(data => {
  const newCategory =  new Category(data);

  return {
    ...newCategory,
    imageUrl: data.imageUrl,
    iconUrl: data.iconUrl,
    created: new Date(data.created),
    updated: new Date(data.updated)
  }
});

// category.js
export default class Category {
   constructor({id = null, parent = null, name = null, description = null, color = null, speechChallenges = null}) {
  }
}
```

That way, creating an Category class, all you have to do is pass an object to the constructor, and the constructor will take in from there.

To consider:
- The object keys MUST match or the variables will have their default value
- Category always has to be created passing an object (that's fine, saves from typing terrible long lines for object creation)